### PR TITLE
use dayjs instead of moment

### PR DESCRIPTION
**Pull request by adrianjost**
# Issue #173 

## Description

- replaced moment.js with day.js. It wasn't 100% compatible but close enough. The on missing function was pretty easy to code in plain js.
- fixed some browser console errors & removed unused code
- added some tests to pass the codecov threshold

### CODE REVIEW

File: src/components/ui/BaseSelect.vue [hunk 1]
```diff
@@ -60,7 +60,7 @@ export default {
 		},
 		optionLabel: {
 			type: String,
-			default: "",
+			default: "label",
```
Comment (adrianjost) [inline src/components/ui/BaseSelect.vue]: add default label that we forgot in #139

File: src/pages/courses/create.vue [hunk 2]
```diff
@@ -86,10 +83,10 @@ export default {
...
 
 			course.times = this.course.times.map((time) => {
-				time.startTime = moment
-					.duration(time.startTime, "HH:mm")
-					.asMilliseconds()
-					.toString();
+				let [startHours, startMinutes] = time.startTime.split(":");
+				startMinutes = startMinutes * 60 * 1000;
+				startHours = startHours * 60 * 60 * 1000;
+				time.startTime = startHours + startMinutes;
```
Comment (adrianjost) [inline src/pages/courses/create.vue]: `.duration` and `.asMilliseconds` is not supported by dayjs 😢 But I think the plain JS version should be fine here.

File: src/pages/teams/_id/members.vue [hunk 3]
```diff
@@ -397,7 +396,7 @@ export default {
 	computed: {
 		invitedMembers() {
 			return this.team.invitedUserIds.map((invite) => {
-				invite.createdAt = moment(invite.createdAt).format("DD.MM.YYYY");
+				invite.createdAt = dayjs(invite.createdAt).format("DD.MM.YYYY");
```
Comment (adrianjost) [inline src/pages/teams/_id/members.vue]: yeah this is compatible 👍

File: tests/unit/setup.js [hunk 4]
```diff
@@ -192,6 +192,13 @@ global.createModuleStore = (vuexModule, options = {}) => {
 };
 */
 
+// ===
+// Let tests fail on console.error
+// ===
+console.error = jest.fn((error) => {
+	throw `💔 ${error}`;
+});
```
Comment (adrianjost) [inline tests/unit/setup.js]: We had some tests that we logging errors to the console, but are not failing. This trick converts all error logs into real errors and let the test fail.

File: src/components/ui/BaseSelect.vue [hunk 5]
```diff
@@ -14,13 +14,12 @@
 			{{ label }}
 		</span>
 		<multi-select
+			v-model="vmodelProxy"
```
Comment (adrianjost) [inline src/components/ui/BaseSelect.vue]: use v-model so we are not relying on the value prop implementation of multi-select. multi-select could now use whatever prop they wan't as a v-model without breaking our code

File: tests/unit/setup.js (hunk 4 above)
Comment (ChrisWeissmann) [reply tests/unit/setup.js]: Nice I was wondering about that while looking at travis earlier :)

**Pull request merged**